### PR TITLE
Baxpr queue handling

### PR DESCRIPTION
**Pull request by baxpr**
@bud42 can you look at this?

Instead of limiting launch based on total number of accre jobs, this will limit based on the number of pending accre jobs. This might help avoid hitting xnat with too many accre job starts at once.

It would also remove the limit on total running jobs, unless we add that in additionally. Not sure what effect that would have, but I think ACCRE's own scheduling will keep that under control just by keeping the pending list full.

If we install this, we would also need to drop the queue_limit setting in the instance dashboard to more like 20-50 instead of 200-500.

### CODE REVIEW

**Comment by bud42** [comment]
Do jobs always enter the pending phase long enough for this to work?
And if so, are we limiting the number of new jobs that launch each time dax runs? Is that what we want?

**Comment by baxpr** [comment]
Ah, I don't know. Probably not.

The goal would be to limit the number of pending jobs to ~50 or whatever avoids too large of a hit when accre launches them all at once.

I'll add back a limit for total jobs as well, I think

**Comment by baxpr** [comment]
So overall, dax will launch as many jobs as possible each hour subject to

- queue_limit total jobs on accre (500 right now)
- number of pending jobs is not too large (say 50)
- slight delay before each launch

All three of these settings can be exposed in the instance redcap.

**Comment by baxpr** [comment]
Could throttle by pending uploads as well, for that matter - no new launches until e.g. pending uploads are <2000

**Comment by baxpr** [comment]
Testing full `dax manager` run on ROGERSTEST (rogersbp@hickory)

**Comment by baxpr** [comment]
Requires new fields `main_queuelimit_pending`, `main_limit_pendinguploads` in the instance redcap. We need to document the instances panel and provide an initial data dictionary for it, similar to the project settings info in `docs/dax_manager.rst`

**Comment by baxpr** [comment]
Tested ok for a full build/launch/upload cycle on a single project, a few assessors. Next, test thresholds

**Comment by baxpr** [comment]
With thresholds set to 1, only 1 job got launched. Would be helpful to report why launching stopped in the log.

**Comment by baxpr** [comment]
Launch delay is working. @bud42 this is ready for another look. Not sure how it will interact with #369 though

File: dax/dax_settings.py [hunk 1]
```diff
@@ -14,7 +14,15 @@
...
 SLURM_JOBTIME = "sacct -j ${jobid}.batch --format CPUTime --noheader"
 
-SLURM_COUNTJOBS = 'squeue -u "$USER" --noheader | wc -l'
+SLURM_COUNTJOBS = 'squeue --me --noheader | wc -l'
+
+SLURM_COUNTJOBS_LAUNCHED = 'squeue --me --noheader | wc -l'
+
+SLURM_COUNTJOBS_PENDING = 'squeue --me -t PENDING --noheader | wc -l'
+
+SLURM_COUNT_PENDINGUPLOADS = 'ls -d /nobackup/$USER/Spider_Upload_Dir/*-x-* |wc -l'
```
Comment (bud42) [inline dax/dax_settings.py:23]: Do we want to assume this path?

File: dax/launcher.py [hunk 2]
```diff
@@ -87,6 +87,9 @@ def __init__(self,
                  yaml_dict=dict(),
                  priority_project=None,
                  queue_limit='10',
+                 queue_limit_pending='10',
```
Comment (bud42) [inline dax/launcher.py:90]: should these defaults be ints?

**Comment by baxpr** [comment]
No. I meant to do that via Template, hang on

**Comment by bud42** [comment]
Looks great! Let's merge this prior to #369. Hopefully, git will work it's magic!

**Comment by baxpr** [comment]
@bud42 see if that should do it?

File: dax/launcher.py (hunk 2 above)
Comment (baxpr) [reply dax/launcher.py:90]: Didn't think about it - I found them this way. I can't quite tell - they get converted here maybe?
https://github.com/VUIIS/dax/blob/2dd32c65eae8e32b6b578b327df17fb552df5971/dax/dax_manager.py#L303

Comment (baxpr) [reply dax/launcher.py]: ...pretty sure they should be ints though so I changed 'em

**Review by bud42 — APPROVED** [review]
looks good!

**Pull request merged**